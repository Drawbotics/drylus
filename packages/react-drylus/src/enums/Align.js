import Enum from '@drawbotics/enums';


const Align = new Enum(
  'LEFT',
  'CENTER',
  'RIGHT',
);


export default Align;
