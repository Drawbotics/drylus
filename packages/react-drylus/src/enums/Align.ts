export enum Align {
  LEFT = 'LEFT',
  CENTER = 'CENTER',
  RIGHT = 'RIGHT',
  JUSTIFY = 'JUSTIFY',
}
