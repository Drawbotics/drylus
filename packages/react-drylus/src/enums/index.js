export Category from './Category';
export Tier from './Tier';
export Size from './Size';
export Position from './Position';
export Align from './Align';
export Color from './Color';
export Shade from './Shade';
