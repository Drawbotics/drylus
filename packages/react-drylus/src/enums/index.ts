export * from './Category';
export * from './Tier';
export * from './Size';
export * from './Position';
export * from './Align';
export * from './Color';
export * from './Shade';
