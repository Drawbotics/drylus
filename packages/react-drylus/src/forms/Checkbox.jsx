import React from 'react';
import v4 from 'uuid/v4';
import { css, cx } from 'emotion';
import sv from '@drawbotics/style-vars';
import PropTypes from 'prop-types';

import Hint from './Hint';
import Icon from '../components/Icon';
import { Sizes } from '../base';
import { getEnumAsClass } from '../utils';


const styles = {
  root: css`
    position: relative;
    display: inline-block;
  `,
  wrapper: css`
    display: inline-flex;
    align-items: center;
    justify-content: flex-start;
    cursor: pointer;

    &:hover {
      [data-element="sprite"] {
        background: ${sv.neutralDark};
      }
    }
  `,
  disabled: css`
    cursor: not-allowed !important;

    > label {
      cursor: not-allowed !important;
      color: ${sv.colorDisabled};
    }
  `,
  error: css`
    [data-element="sprite"] {
      box-shadow: inset 0px 0px 0px 2px ${sv.red};
    }

    [data-element="icon"] {
      background: ${sv.red};
    }
  `,
  checkbox: css`
    height: ${sv.defaultMargin};
    width: ${sv.defaultMargin};
    position: relative;
    overflow: hidden;
  `,
  input: css`
    visibility: hidden;

    &:checked + [data-element="sprite"] {
      [data-element="icon"] {
        transform: scale(1);
        border-radius: 0;
      }
    }

    &:disabled + [data-element="sprite"] {
      cursor: not-allowed;
      background: ${sv.neutralLight} !important;

      [data-element="icon"] {
        opacity: 0.7;
      }
    }
  `,
  label: css`
    margin-left: ${sv.marginExtraSmall};
    color: ${sv.colorPrimary};

    &:hover {
      cursor: pointer;
    }
  `,
  sprite: css`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: ${sv.defaultBorderRadius};
    background: ${sv.neutral};
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    cursor: pointer;
    transition: ${sv.defaultTransition};
    overflow: hidden;
  `,
  iconLabel: css`
    cursor: inherit;
    color: ${sv.white};
    height: 100%;
    width: 100%;
    background: ${sv.green};
    line-height: ${sv.marginLarge};
    transform: scale(0);
    transition: all ${sv.transitionTimeShort} ${sv.bouncyTransitionCurve};
    border-radius: 100px;

    > i {
      font-size: 1.1rem;
    }
  `,
  small: css`
    > div {
      height: ${sv.marginSmall};
      width: ${sv.marginSmall};
    }
    [data-element="icon"] {
      line-height: calc(${sv.marginSmall} + 1px);

       > i {
         font-size: 0.7rem;
       }
    }

    [data-element="label"] {
      font-size: 0.9rem;
    }
  `,
};



const Checkbox = ({
  onChange,
  value,
  id,
  children,
  disabled,
  error,
  size,
  ...rest,
}) => {
  const isChecked = !! value;

  const handleOnChange = (e) => {
    e.stopPropagation();
    onChange ? onChange( ! isChecked, e.target.name) : null;
  };

  const uniqId = id ? id : v4();
  return (
    <div className={styles.root}>
      <label className={cx(styles.wrapper, {
        [styles[getEnumAsClass(size)]]: size,
        [styles.disabled]: disabled,
        [styles.error]: error,
      })} htmlFor={uniqId}>
        <div className={styles.checkbox}>
          <input
            disabled={disabled}
            checked={isChecked}
            id={uniqId}
            type="checkbox"
            className={styles.input}
            onChange={handleOnChange}
            {...rest} />
          <div data-element="sprite" className={styles.sprite}>
            <label data-element="icon" className={styles.iconLabel} htmlFor={uniqId}>
              <Icon bold name="check" />
            </label>
          </div>
        </div>
        {do{
          if (children) {
            <label data-element="label" className={styles.label} htmlFor={uniqId}>
              {children}
            </label>
          }
        }}
      </label>
      {do{
        if (error && typeof error === 'string') {
          <Hint error>{error}</Hint>
        }
      }}
    </div>
  );
};


Checkbox.propTypes = {
  /** If passed, the text will be the label of the checkbox */
  children: PropTypes.string,

  /** Triggered when checkbox value is changed */
  onChange: PropTypes.func,

  /** If true, checkbox is not clickable */
  disabled: PropTypes.bool,

  /** Determines if checkbox is checked */
  value: PropTypes.bool,

  /** Error text to prompt the user to act, or a boolean if you don't want to show a message */
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  /** Size of the checkbox. Can be small or default */
  size: PropTypes.oneOf([Sizes.SMALL, Sizes.DEFAULT]),
};


Checkbox.defaultProps = {
  size: Sizes.DEFAULT,
};


export default Checkbox;
