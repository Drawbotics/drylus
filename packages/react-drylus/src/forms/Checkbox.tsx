import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';
import { v4 } from 'uuid';

import { Icon } from '../components';
import { placeholderStyles } from '../components';
import { Category, Size } from '../enums';
import { Responsive, Style } from '../types';
import { getEnumAsClass, run, useResponsiveProps } from '../utils';
import { Hint } from './Hint';

const styles = {
  root: css`
    position: relative;
    display: inline-block;
  `,
  wrapper: css`
    display: inline-flex;
    align-items: center;
    justify-content: flex-start;
    cursor: pointer;

    &:hover {
      [data-element='sprite'] {
        background: ${sv.neutralDark};
      }
    }
  `,
  readOnly: css`
    pointer-events: none;

    [data-element='sprite'] {
      background: ${sv.neutralLight} !important;
    }

    [data-element='locked-icon'] {
      transform: scale(1);
      border-radius: 0;
      background: none;
      color: ${sv.colorSecondary};
    }
  `,
  disabled: css`
    cursor: not-allowed !important;

    > label {
      cursor: not-allowed !important;
      color: ${sv.colorDisabled};
    }
  `,
  error: css`
    [data-element='sprite'] {
      box-shadow: inset 0px 0px 0px 2px ${sv.red};
    }

    [data-element='icon'] {
      background: ${sv.red};
    }
  `,
  checkbox: css`
    height: calc(${sv.defaultMargin} - 4px);
    width: calc(${sv.defaultMargin} - 4px);
    position: relative;
    overflow: hidden;
  `,
  input: css`
    visibility: hidden;

    &:checked + [data-element='sprite'] {
      [data-element='icon'] {
        transform: scale(1);
        border-radius: 0;
      }
      [data-element='locked-icon'] {
        background: ${sv.green};
        color: ${sv.colorPrimaryInverse};
      }
    }

    &:disabled + [data-element='sprite'] {
      cursor: not-allowed;
      background: ${sv.neutralLight} !important;

      [data-element='icon'] {
        opacity: 0.7;
      }
    }
  `,
  label: css`
    margin-left: ${sv.marginExtraSmall};
    color: ${sv.colorPrimary};
    position: relative;
    top: 1px;

    &:hover {
      cursor: pointer;
    }
  `,
  sprite: css`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: ${sv.defaultBorderRadius};
    background: ${sv.neutral};
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    cursor: pointer;
    transition: ${sv.transitionShort};
    overflow: hidden;
  `,
  iconLabel: css`
    cursor: inherit;
    color: ${sv.white};
    height: 100%;
    width: 100%;
    background: ${sv.green};
    line-height: ${sv.defaultMargin};
    transform: scale(0);
    transition: all ${sv.transitionTimeShort} ${sv.bouncyTransitionCurve};
    border-radius: 100px;

    > i {
      font-size: 0.8rem;
    }
  `,
  large: css`
    > div {
      height: ${sv.marginLarge};
      width: ${sv.marginLarge};
    }

    [data-element='label'] {
      font-size: 1.1rem;
      margin-left: ${sv.marginSmall};
      top: 0;
    }

    [data-element='icon'],
    [data-element='locked-icon'] {
      line-height: calc(${sv.marginLarge} + 5px);

      > i {
        font-size: 1.2rem !important;
      }
    }
  `,
};

interface CheckboxProps {
  /** The dom property */
  id?: string;

  /** If passed, the text will be the label of the checkbox */
  children?: string;

  /** Triggered when checkbox value is changed */
  onChange?: (value: boolean, name?: string) => void;

  /** If true, checkbox is not clickable */
  disabled?: boolean;

  /** Determines if checkbox is checked */
  value?: boolean;

  /** Name of the form element (target.name) */
  name?: string;

  /** Error text to prompt the user to act, or a boolean if you don't want to show a message */
  error?: string | boolean;

  /**
   * Size of the checkbox. Can be large or default
   * @default Size.DEFAULT
   */
  size?: Exclude<
    Size,
    Size.EXTRA_SMALL | Size.SMALL | Size.EXTRA_LARGE | Size.HUGE | Size.EXTRA_HUGE | Size.MASSIVE
  >;

  /** If true, a loading overlay is displayed on top of the component */
  isPlaceholder?: boolean;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;

  /** @private */
  [x: string]: any;
}

export const Checkbox = ({ responsive, ...rest }: CheckboxProps) => {
  const {
    onChange,
    value,
    id,
    children,
    disabled,
    error,
    size = Size.DEFAULT,
    style,
    isPlaceholder,
    ...props
  } = useResponsiveProps<CheckboxProps>(rest, responsive);

  const isChecked = value === true;

  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
    e.stopPropagation();
    onChange ? onChange(!isChecked, (e.target as HTMLInputElement).name) : null;
  };

  const uniqId = id ? id : v4();
  const readOnly = onChange == null;
  return (
    <div style={style} className={styles.root}>
      <label
        className={cx(styles.wrapper, {
          [styles[getEnumAsClass<typeof styles>(size)]]: size != null,
          [styles.disabled]: disabled === true,
          [styles.error]: error != null && error !== false,
          [styles.readOnly]: readOnly,
          [placeholderStyles.shimmer]: isPlaceholder,
        })}
        htmlFor={uniqId}>
        <div className={styles.checkbox}>
          <input
            disabled={disabled}
            checked={isChecked}
            id={uniqId}
            type="checkbox"
            className={styles.input}
            onChange={handleOnChange}
            readOnly={readOnly}
            {...props}
          />
          <div data-element="sprite" className={styles.sprite}>
            {run(() => {
              if (readOnly) {
                return (
                  <label data-element="locked-icon" className={styles.iconLabel}>
                    <Icon name="lock" />
                  </label>
                );
              } else {
                return (
                  <label data-element="icon" className={styles.iconLabel} htmlFor={uniqId}>
                    <Icon bold name="check" />
                  </label>
                );
              }
            })}
          </div>
        </div>
        {run(() => {
          if (children) {
            return (
              <label data-element="label" className={styles.label} htmlFor={uniqId}>
                {children}
              </label>
            );
          }
        })}
      </label>
      {run(() => {
        if (error && typeof error === 'string') {
          return <Hint category={Category.DANGER}>{error}</Hint>;
        }
      })}
    </div>
  );
};
