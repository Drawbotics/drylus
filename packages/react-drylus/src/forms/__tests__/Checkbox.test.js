import React from 'react';
import { create } from 'react-test-renderer';

import Checkbox from '../Checkbox';
import { Size } from '../../enums';


jest.mock('uuid/v4', () => {
  return jest.fn(() => 1);
});



describe('Checkbox', () => {
  const onChange = jest.fn();

  describe('matches snapshot when', () => {
    it('is not checked', () => {
      const tree = create(
        <Checkbox onChange={onChange} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is checked', () => {
      const tree = create(
        <Checkbox value={true} onChange={onChange} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a label', () => {
      const tree = create(
        <Checkbox onChange={onChange}>
          Label
        </Checkbox>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is read only', () => {
      const tree = create(
        <Checkbox value={true} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is large', () => {
      const tree = create(
        <Checkbox
          size={Size.LARGE}
          onChange={onChange} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  describe('changes state when clicked', () => {
    let checked = false;

    expect(checked).toBeFalsy();

    const component = create(
      <Checkbox
        value={checked}
        onChange={(c) => checked = c}>
        Label
      </Checkbox>
    );

    const input = component.root.findByProps({ type: 'checkbox' });
    const changeEvent = new Event('change');
    Object.defineProperty(changeEvent, 'target', { writable: false, value: input });

    input.props.onChange(changeEvent);

    expect(checked).toBeTruthy();
  });
});
