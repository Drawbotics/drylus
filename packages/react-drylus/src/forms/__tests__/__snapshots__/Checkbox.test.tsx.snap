// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox matches snapshot when has a label 1`] = `
.emotion-7 {
  position: relative;
  display: inline-block;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-6:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-5 {
  margin-left: 8px;
  color: #172b4e;
  position: relative;
  top: 1px;
}

.emotion-5:hover {
  cursor: pointer;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-1 {
  color: inherit;
  font-weight: bold !important;
}

<div
  className="emotion-7"
>
  <label
    className="emotion-6 undefined"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={false}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="icon"
          htmlFor={1}
        >
          <i
            className="Drycon Drycon-check emotion-1"
          />
        </label>
      </div>
    </div>
    <label
      className="emotion-5"
      data-element="label"
      htmlFor={1}
    >
      Label
    </label>
  </label>
</div>
`;

exports[`Checkbox matches snapshot when has a label as react node 1`] = `
.emotion-8 {
  position: relative;
  display: inline-block;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-7:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-6 {
  margin-left: 8px;
  color: #172b4e;
  position: relative;
  top: 1px;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-1 {
  color: inherit;
  font-weight: bold !important;
}

.emotion-5 {
  display: inline;
  font-weight: 500;
  -webkit-letter-spacing: 0.01rem;
  -moz-letter-spacing: 0.01rem;
  -ms-letter-spacing: 0.01rem;
  letter-spacing: 0.01rem;
  color: #172b4e;
  font-size: 1rem;
}

<div
  className="emotion-8"
>
  <label
    className="emotion-7 undefined"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={false}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="icon"
          htmlFor={1}
        >
          <i
            className="Drycon Drycon-check emotion-1"
          />
        </label>
      </div>
    </div>
    <label
      className="emotion-6"
      data-element="label"
      htmlFor={1}
    >
      <span
        className="emotion-5"
      >
        Label
      </span>
    </label>
  </label>
</div>
`;

exports[`Checkbox matches snapshot when is checked 1`] = `
.emotion-6 {
  position: relative;
  display: inline-block;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-5:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-1 {
  color: inherit;
  font-weight: bold !important;
}

<div
  className="emotion-6"
>
  <label
    className="emotion-5 undefined"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={true}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="icon"
          htmlFor={1}
        >
          <i
            className="Drycon Drycon-check emotion-1"
          />
        </label>
      </div>
    </div>
  </label>
</div>
`;

exports[`Checkbox matches snapshot when is large 1`] = `
.emotion-6 {
  position: relative;
  display: inline-block;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-1 {
  color: inherit;
  font-weight: bold !important;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-5:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-5 > div {
  height: 24px;
  width: 24px;
}

.emotion-5 [data-element='icon'],
.emotion-5 [data-element='locked-icon'] {
  line-height: calc(24px + 5px);
}

.emotion-5 [data-element='icon'] > i,
.emotion-5 [data-element='locked-icon'] > i {
  font-size: 1rem !important;
}

<div
  className="emotion-6"
>
  <label
    className="emotion-5"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={false}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="icon"
          htmlFor={1}
        >
          <i
            className="Drycon Drycon-check emotion-1"
          />
        </label>
      </div>
    </div>
  </label>
</div>
`;

exports[`Checkbox matches snapshot when is not checked 1`] = `
.emotion-6 {
  position: relative;
  display: inline-block;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-5:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-1 {
  color: inherit;
  font-weight: bold !important;
}

<div
  className="emotion-6"
>
  <label
    className="emotion-5 undefined"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={false}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={false}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="icon"
          htmlFor={1}
        >
          <i
            className="Drycon Drycon-check emotion-1"
          />
        </label>
      </div>
    </div>
  </label>
</div>
`;

exports[`Checkbox matches snapshot when is read only 1`] = `
.emotion-1 {
  color: inherit;
}

.emotion-6 {
  position: relative;
  display: inline-block;
}

.emotion-4 {
  height: 16px;
  width: 16px;
  position: relative;
  overflow: hidden;
}

.emotion-0 {
  visibility: hidden;
}

.emotion-0:checked + [data-element='sprite'] [data-element='icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
}

.emotion-0:checked + [data-element='sprite'] [data-element='locked-icon'] {
  background: #3FBC9B;
  color: #fff;
}

.emotion-0:disabled + [data-element='sprite'] {
  cursor: not-allowed;
  background: #eaeff4 !important;
}

.emotion-3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 2px;
  background: #d3dce6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-2 {
  cursor: inherit;
  color: #fff;
  height: 100%;
  width: 100%;
  background: #3FBC9B;
  line-height: calc(16px + 2px);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  transition: all 0.15s cubic-bezier(0.44,0.11,0.07,1.29);
  border-radius: 100px;
}

.emotion-2 > i {
  font-size: 0.8rem;
  margin-left: 1px;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  pointer-events: none;
}

.emotion-5:hover [data-element='sprite'] {
  background: #879ba6;
}

.emotion-5 [data-element='sprite'] {
  background: #eaeff4 !important;
}

.emotion-5 [data-element='locked-icon'] {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  border-radius: 0;
  background: none;
  color: #879ba6;
}

<div
  className="emotion-6"
>
  <label
    className="undefined emotion-5"
    htmlFor={1}
  >
    <div
      className="emotion-4"
    >
      <input
        checked={true}
        className="emotion-0"
        id={1}
        onChange={[Function]}
        readOnly={true}
        type="checkbox"
      />
      <div
        className="emotion-3"
        data-element="sprite"
      >
        <label
          className="emotion-2"
          data-element="locked-icon"
        >
          <i
            className="emotion-1 Drycon Drycon-lock"
          />
        </label>
      </div>
    </div>
  </label>
</div>
`;
